packages/catalog-process/test/api/createEservice.test.ts

```diff
@@ -2,166 +2,89 @@
   };
 
-  const mockApiEservice: catalogApi.EService =
-    eServiceToApiEService(mockEService);
-
-  const mockEserviceSeed: EServiceSeed = {
-    name: mockApiEservice.name,
-    description: mockApiEservice.description,
-    technology: "REST",
-    mode: "RECEIVE",
-    descriptor: {
-      audience: mockApiEservice.descriptors[0].audience,
-      voucherLifespan: mockApiEservice.descriptors[0].voucherLifespan,
-      dailyCallsPerConsumer:
-        mockApiEservice.descriptors[0].dailyCallsPerConsumer,
-      dailyCallsTotal: mockApiEservice.descriptors[0].dailyCallsTotal,
-      agreementApprovalPolicy:
-        mockApiEservice.descriptors[0].agreementApprovalPolicy,
-    },
-  };
-
-  vi.spyOn(catalogService, "createEService").mockImplementation(() =>
-    Promise.resolve(mockEService)
+  const mockApiEservice: catalogApi.EService = catalogApi.EService.parse(
+    eServiceToApiEService(mockEService)
   );
 
-  const authData: AuthData = {
-    ...getMockAuthData(),
-    userRoles: ["admin", "api"],
-  };
+  const mockEserviceSeed = getMockEserviceSeed(mockApiEservice);
 
-  const validToken = jwt.sign(createPayload(authData), "test-secret");
+  vi.spyOn(catalogService, "createEService").mockResolvedValue(mockEService);
 
-  it("Should return 200 for a user with ADMIN_ROLE, API_ROLE role", async () => {
-    const res = await request(api)
+  const generateToken = (authData: AuthData) =>
+    jwt.sign(createPayload(authData), "test-secret");
+
+  const makeRequest = async (token: string, payload: object) =>
+    request(api)
       .post("/eservices")
-      .set("Authorization", `Bearer ${validToken}`)
+      .set("Authorization", `Bearer ${token}`)
       .set("X-Correlation-Id", generateId())
-      .send(mockEserviceSeed);
-
-    expect(res.status).toBe(200);
-    expect(res.body).toEqual(mockApiEservice);
-  });
+      .send(payload);
 
-  it("Should return 403 for a user with role other than ADMIN_ROLE, API_ROLE", async () => {
-    const invalidAuthData: AuthData = {
-      ...getMockAuthData(),
-      userRoles: ["internal"],
-    };
+  it.each([userRoles.ADMIN_ROLE, userRoles.API_ROLE])(
+    "Should return 200 for user with role %s",
+    async (role) => {
+      const token = generateToken({ ...getMockAuthData(), userRoles: [role] });
+      const res = await makeRequest(token, mockEserviceSeed);
 
-    const invalidToken = jwt.sign(
-      createPayload(invalidAuthData),
-      "test-secret"
-    );
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(mockApiEservice);
+    }
+  );
 
-    const res = await request(api)
-      .post("/eservices")
-      .set("Authorization", `Bearer ${invalidToken}`)
-      .set("X-Correlation-Id", generateId())
-      .send(mockEserviceSeed);
+  it.each([
+    userRoles.INTERNAL_ROLE,
+    userRoles.M2M_ROLE,
+    userRoles.MAINTENANCE_ROLE,
+    userRoles.SECURITY_ROLE,
+    userRoles.SUPPORT_ROLE,
+  ])("Should return 403 for user with role %s", async (role) => {
```

Comment:
```suggestion
  it.each(Object.values(userRoles).filter(
  (role) => role !== userRoles.ADMIN_ROLE && role !== userRoles.API_ROLE
))("Should return 403 for user with role %s", async (role) => {
```
It's better to derive the roles from the entire set

Reply:
Great tip